Source/Events.Store.MongoDB/Processing/Streams/StreamProcessorStateRepository.cs

```diff
@@ -33,13 +33,19 @@ public StreamProcessorStateRepository(EventStoreConnection connection, ILogger l
         }
 
         /// <inheritdoc/>
-        public async Task<Runtime.Events.Processing.Streams.StreamProcessorState> GetOrAddNew(Runtime.Events.Processing.Streams.StreamProcessorId streamProcessorId, CancellationToken cancellationToken)
+        public async Task<Runtime.Events.Processing.Streams.StreamProcessorState> GetOrAddNew(
+            Runtime.Events.Processing.Streams.StreamProcessorId streamProcessorId,
+            CancellationToken cancellationToken)
         {
             try
             {
-                var states = _connection.StreamProcessorStates;
+                var states = await _connection.GetStreamProcessorStateCollection(streamProcessorId.ScopeId, cancellationToken)
+                    .ConfigureAwait(false);
```

Comment:
I would keep them on the same line 
```suggestion
                var states = await _connection.GetStreamProcessorStateCollection(streamProcessorId.ScopeId, cancellationToken).ConfigureAwait(false);
```